src/app/attendance/actions/getAttendance.ts

```diff
@@ -0,0 +1,30 @@
+"use server";
+
+import { Tables } from "@/lib/db";
+import { getSupabaseClient } from "../../../../utils/supabase/supabaseClient";
+import { getMembers } from "@/app/members/actions/getMembers";
+
+export const getAttendance = async () => {
+  const supabase = await getSupabaseClient();
+
+  const membersResponse = await getMembers();
+  const members = membersResponse?.data || [];
+
+  const { data: attendanceData } = await supabase
+    .from(Tables.Attendance)
+    .select("*");
+
+  const enrichedData = (attendanceData || []).map((attendance) => {
+    const member = members.find((m) => m.id === attendance.memberId);
+
+    return {
+      ...attendance,
+      name: member ? member.name : "Unknown",
+      shift: member ? member.shift : "Unknown Shift",
```

Comment:
@Mubashar1009  Why is it necessary to make the name and shift conditional? And why should we use unknown as the name and for shift?

Reply:
fixed